app/src/test/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java

```diff
@@ -0,0 +1,276 @@
+        StubCommunityService.groups.add(new Group("Elephants", "Enric",
+            DateTime.Builder.buildDateString("2020-04-14")));
+        StubCommunityService.groups.get(HUSKY).addSubscriber(new User("John Smith",
+            "johnSmith@gmail.com", "1234"));
+        StubCommunityService.groups.add(new Group("Dinosaur", "Gradle",
+            DateTime.Builder.buildDateString("2019-11-23")));
+    }
+
+    @Override
+    public SortedSet<Group> getAllGroups() {
+        return new TreeSet<>(groups);
+    }
+
+    @Override
+    public void createGroup(User user, Group group) throws GroupAlreadyExistingException {
+        for (Group actualGroup : groups) {
+            if (group.equals(actualGroup)) {
+                throw new GroupAlreadyExistingException();
+            }
+        }
+        groups.add(group);
+    }
+
+    @Override
+    public void deleteGroup(String groupName) throws GroupNotFoundException {
+        boolean found = false;
+        for (int i = 0; i < groups.size() && !found; ++i) {
+            if (groupName.equals(groups.get(i).getName())) {
+                groups.remove(i);
+                found = true;
+            }
+        }
+        if (!found) {
+            throw new GroupNotFoundException();
+        }
+    }
+
+    @Override
+    public boolean isGroupExisting(Group group) {
+        return groups.contains(group);
+    }
+
+    @Override
+    public void addSubscriber(User user, Group group) {
+        int index = groups.indexOf(group);
+        user.addSubscribedGroup(groups.get(index));
+    }
+
+    @Override
+    public void deleteSubscriber(User user, Group group) {
+        int index = groups.indexOf(group);
+        groups.get(index).removeSubscriber(user);
+    }
+
+    @Override
+    public void createForum(User user, Group group, Forum forum) {
+        int index = groups.indexOf(group);
+        groups.get(index).addForum(forum);
+    }
+
+    @Override
+    public void deleteForum(User user, Group group, Forum forum) throws ForumNotFoundException, NotForumOwnerException {
+        if (!forumExists(group, forum)) {
+            throw new ForumNotFoundException();
+        }
+        if (!forum.getOwnerUsername().equals(user.getUsername())) {
+            throw new NotForumOwnerException();
+        }
+        group.removeForum(forum);
+    }
+
+    /**
+     * Checks whether a forum exists or not.
+     * @param group The group where the forum should be.
+     * @param forum The forum to check
+     * @return True if the forum was found or false otherwise
+     */
+    private boolean forumExists(Group group, Forum forum) {
+        for (Group g : groups) {
+            if (g.getName().equals(group.getName())) {
+                for (Forum f : g.getForums()) {
+                    if (f.getName().equals(forum.getName())) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    @Override
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method length is 24 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5426840)